target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
OK, I fixed the flash driver, I am now able to flash and boot the system:

```
### RS353aw (Hardware-Rev. 1.0, Firmware-Rev. 1.1) ###

CPU   Check ... passed (MIPS 34KEc [MT] @ 500/250.0 MHz)
SDRAM Check ................................................................................................................................. passed (128 MByte)
FLASH Check  passed (32 MByte)

### Selftest passed ###


Press <sp> for boot monitor or any other key to boot system

Booting Image from Flash ROM
Checking image ... failed: CRC-error (0x59d75e00 <> 0xbb5527b9)
OK
Writing image to RAM (Release 42.1) . OK (131088 bytes)
Booting BOSS...
boss image started at 0x2d6a0
Disabling WDT


OpenWrt kernel loader for MIPS based SoC
Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>
Decompressing kernel... done!
Starting kernel at 80002000...

[    0.000000] Linux version 4.4.14 (qa@serveurQA) (gcc version 5.3.0 (OpenWrt GCC 5.3.0 50020) ) #61 SMP Wed Jan 11 15:34:50 UTC 2017
[    0.000000] SoC: xRX200 rev 1.2
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019556 (MIPS 34Kc)
[    0.000000] MIPS: machine is RS353 - Bintec Elmeg RS353
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Detected 1 available secondary CPU(s)
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] PERCPU: Embedded 9 pages/cpu @8110c000 s8128 r8192 d20544 u36864
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyLTQ0,115200 init=/etc/preinit
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=00048501
[    0.000000] Readback ErrCtl register=00048501
[    0.000000] Memory: 122908K/131072K available (4028K kernel code, 175K rwdata, 1172K rodata, 1264K init, 222K bss, 8164K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000] NR_IRQS:256
[    0.000000] Setting up vectored interrupts
[    0.000000] Setting up IPI vectored interrupts
[    0.000000] CPU Clock: 500MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041786 ns
[    0.000011] sched_clock: 32 bits at 250MHz, resolution 4ns, wraps every 8589934590ns
[    0.007840] Calibrating delay loop... 332.54 BogoMIPS (lpj=665088)
[    0.042408] pid_max: default: 32768 minimum: 301
[    0.047259] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.053806] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.064882] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.064896] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.065118] CPU1 revision is: 00019556 (MIPS 34Kc)
[    0.124658] Synchronize counters for CPU 1: done.
[    0.129542] Brought up 2 CPUs
[    0.136922] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.146784] pinctrl core: initialized pinctrl subsystem
[    0.152759] NET: Registered protocol family 16
[    0.163260] pinctrl-xway 1e100b10.pinmux: Init done
[    0.169336] dma-xway 1e104100.dma: Init done - hw rev: 7, ports: 7, channels: 28
[    0.282496] dcdc-xrx200 1f106a00.dcdc: Core Voltage : 1016 mV
[    0.445381] usbcore: registered new interface driver usbfs
[    0.451125] usbcore: registered new interface driver hub
[    0.456605] usbcore: registered new device driver usb
[    0.462381] PCI host bridge to bus 0000:00
[    0.466496] pci_bus 0000:00: root bus resource [mem 0x1c000000-0x1cffffff]
[    0.473399] pci_bus 0000:00: root bus resource [io  0x1d800000-0x1d8fffff]
[    0.480320] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.487133] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.495173] ifx_pcie_rc_class_early_fixup: fixed pcie host bridge to pci-pci bridge
[    0.512346] pci 0000:00:00.0: BAR 8: assigned [mem 0x1c000000-0x1c0fffff]
[    0.519102] pci 0000:00:00.0: BAR 9: assigned [mem 0x1c100000-0x1c1fffff pref]
[    0.526362] pci 0000:01:00.0: BAR 0: assigned [mem 0x1c000000-0x1c01ffff 64bit]
[    0.533724] pci 0000:01:00.0: BAR 6: assigned [mem 0x1c100000-0x1c10ffff pref]
[    0.540946] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.545965] pci 0000:00:00.0:   bridge window [mem 0x1c000000-0x1c0fffff]
[    0.552788] pci 0000:00:00.0:   bridge window [mem 0x1c100000-0x1c1fffff pref]
[    0.560065] ifx_pcie_bios_map_irq port 0 dev 0000:00:00.0 slot 0 pin 1 
[    0.566677] ifx_pcie_bios_map_irq dev 0000:00:00.0 irq 144 assigned
[    0.573026] ifx_pcie_bios_map_irq port 0 dev 0000:01:00.0 slot 0 pin 1 
[    0.579637] ifx_pcie_bios_map_irq dev 0000:01:00.0 irq 144 assigned
[    0.587669] clocksource: Switched to clocksource MIPS
[    0.595453] NET: Registered protocol family 2
[    0.600770] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.607724] TCP bind hash table entries: 1024 (order: 1, 8192 bytes)
[    0.614097] TCP: Hash tables configured (established 1024 bind 1024)
[    0.620590] UDP hash table entries: 256 (order: 1, 8192 bytes)
[    0.626413] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes)
[    0.633028] NET: Registered protocol family 1
[    0.637963] gptu: totally 6 16-bit timers/counters
[    0.642864] gptu: misc_register on minor 63
[    0.647078] gptu: succeeded to request irq 126
[    0.651562] gptu: succeeded to request irq 127
[    0.656057] gptu: succeeded to request irq 128
[    0.660550] gptu: succeeded to request irq 129
[    0.665057] gptu: succeeded to request irq 130
[    0.669534] gptu: succeeded to request irq 131
[    0.675718] phy-xrx200 gphy-xrx200: requesting lantiq/vr9_phy11g_a2x.bin
[    0.683216] phy-xrx200 gphy-xrx200: booting GPHY0 firmware at 7A00000
[    0.689640] phy-xrx200 gphy-xrx200: booting GPHY1 firmware at 7A00000
[    0.774753] futex hash table entries: 512 (order: 2, 16384 bytes)
[    0.798640] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.804487] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.817625] io scheduler noop registered
[    0.821544] io scheduler deadline registered (default)
[    0.827849] 1e100c00.serial: ttyLTQ0 at MMIO 0x1e100c00 (irq = 112, base_baud = 0) is a lantiq,asc
[    0.836799] console [ttyLTQ0] enabled
[    0.836799] console [ttyLTQ0] enabled
[    0.844143] bootconsole [early0] disabled
[    0.844143] bootconsole [early0] disabled
[    0.854007] lantiq nor flash device: 02000000 at 10000000
[    0.859419] ltq_nor: Found 1 x16 devices at 0x0 in 16-bit bank. Manufacturer ID 0x0000c2 Chip ID 0x00227e
[    0.868912] Amd/Fujitsu Extended Query Table at 0x0040
[    0.874043]   Amd/Fujitsu Extended Query version 1.3.
[    0.879082] number of CFI chips: 1
[    0.882560] 8 ofpart partitions found on MTD device ltq_nor
[    0.888047] Creating 8 MTD partitions on "ltq_nor":
[    0.892929] 0x000000000000-0x000000020000 : "logic_config"
[    0.900651] 0x000000020000-0x000000040000 : "logic_addr"
[    0.908133] 0x000000040000-0x000000060000 : "bootmonitor"
[    0.915529] 0x000000080000-0x0000000c0000 : "flash_config"
[    0.923036] 0x0000000c0000-0x000000100000 : "uboot"
[    0.930065] 0x000000100000-0x000000120000 : "uboot-env"
[    0.937527] 0x000000120000-0x000001fe0000 : "firmware"
[    0.956690] 2 uimage-fw partitions found on MTD device firmware
[    0.962567] 0x000000120000-0x0000002d3a42 : "kernel"
[    0.969436] 0x0000002d3a42-0x000001fe0000 : "rootfs"
[    0.976832] mtd: device 8 (rootfs) set to be root filesystem
[    0.982463] 1 squashfs-split partitions found on MTD device rootfs
[    0.988600] 0x000000560000-0x000001fe0000 : "rootfs_data"
[    0.996215] 0x000001fe0000-0x000002000000 : "GPHY_CPR"
[    1.111392] libphy: lantiq,xrx200-mdio: probed
[    1.185403] eth0: attached PHY [Lantiq XWAY VR9 GPHY 11G v1.4] (phy_addr=0:13, irq=-1)
[    1.257398] eth0: attached PHY [Lantiq XWAY VR9 GPHY 11G v1.4] (phy_addr=0:11, irq=-1)
[    1.329452] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:00, irq=-1)
[    1.401424] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:01, irq=-1)
[    1.473425] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:02, irq=-1)
[    1.482157] wdt 1f8803f0.watchdog: Init done
[    1.489406] NET: Registered protocol family 10
[    1.498407] NET: Registered protocol family 17
[    1.502887] bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need this.
[    1.515445] 8021q: 802.1Q VLAN Support v1.8
[    1.529620] VFS: Mounted root (squashfs filesystem) readonly on device 31:8.
[    1.539336] Freeing unused kernel memory: 1264K (80544000 - 80680000)
[    2.754648] init: Console is alive
[    2.757040] init: - watchdog -
[    3.436125] eth0: port 0 got link
[    4.132920] dwc2 1e101000.ifxhcd: Configuration mismatch. Forcing host mode
[    4.995953] dwc2 1e101000.ifxhcd: DWC OTG Controller
[    4.999538] dwc2 1e101000.ifxhcd: new USB bus registered, assigned bus number 1
[    5.006879] dwc2 1e101000.ifxhcd: irq 62, io mem 0x00000000
[    5.012394] dwc2 1e101000.ifxhcd: Hardware does not support descriptor DMA mode -
[    5.019847] dwc2 1e101000.ifxhcd: falling back to buffer DMA mode.
[    5.027762] hub 1-0:1.0: USB hub found
[    5.030368] hub 1-0:1.0: 1 port detected
[    5.034748] dwc2 1e106000.ifxhcd: Configuration mismatch. Forcing host mode
[    5.895944] dwc2 1e106000.ifxhcd: DWC OTG Controller
[    5.899509] dwc2 1e106000.ifxhcd: new USB bus registered, assigned bus number 2
[    5.906880] dwc2 1e106000.ifxhcd: irq 91, io mem 0x00000000
[    5.912388] dwc2 1e106000.ifxhcd: Hardware does not support descriptor DMA mode -
[    5.919834] dwc2 1e106000.ifxhcd: falling back to buffer DMA mode.
[    5.927492] hub 2-0:1.0: USB hub found
[    5.929991] hub 2-0:1.0: 1 port detected
[    5.940747] init: - preinit -
[    6.355584] random: procd urandom read with 78 bits of entropy available
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    9.656152] jffs2_scan_eraseblock(): End of filesystem marker found at 0x20000
[    9.662050] jffs2_build_filesystem(): unlocking the mtd device... done.
[    9.668540] jffs2_build_filesystem(): erasing all blocks after the end marker... 
[   12.607666] random: nonblocking pool is initialized
[  128.338254] done.
[  128.338768] jffs2: notice: (423) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[  128.356478] mount_root: overlay filesystem has not been fully initialized yet
[  128.363885] mount_root: switching to jffs2 overlay
- config restore -
[  128.565648] procd: - early -
[  128.567247] procd: - watchdog -
[  129.305814] procd: - ubus -
[  129.360584] procd: - init -
Please press Enter to activate this console.
[  130.191538] IFXOS, Version 1.5.19 (c) Copyright 2009, Lantiq Deutschland GmbH
[  130.200977] NET: Registered protocol family 8
[  130.204015] NET: Registered protocol family 20
[  130.214206] PPP generic driver version 2.4.2
[  130.223405] ip6_tables: (C) 2000-2006 Netfilter Core Team
[  130.246312] Lantiq (VRX) DSL CPE MEI driver, version 1.4.8.5, (c) 2013 Lantiq Deutschland GmbH

Lantiq CPE API Driver version: DSL CPE API V4.16.6.3
[  130.266512] 
[  130.266512] Predefined debug level: 3
[  130.277178] Loading modules backported from Linux version wt-2016-05-12-0-g7a54796
[  130.283367] Backport generated by backports.git backports-20160216-0-ge3c56e4
[  130.293391] ip_tables: (C) 2000-2006 Netfilter Core Team
[  130.303397] Infineon Technologies DEU driver version 2.0.0 
[  130.309143] IFX DEU DES initialized (multiblock).
[  130.313804] IFX DEU AES initialized (multiblock).
[  130.317782] IFX DEU ARC4 initialized (multiblock).
[  130.322230] IFX DEU SHA1 initialized.
[  130.325979] IFX DEU MD5 initialized.
[  130.329445] IFX DEU SHA1_HMAC initialized.
[  130.333567] IFX DEU MD5_HMAC initialized.
[  130.344408] nf_conntrack version 0.5.0 (1940 buckets, 7760 max)
[  130.402217] NET: Registered protocol family 24
[  130.421753] xt_time: kernel timezone is -0000
[  130.518249] PCI: Enabling device 0000:00:00.0 (0000 -> 0002)
[  130.522605] PCI: Enabling device 0000:01:00.0 (0000 -> 0002)
[  130.655972] ieee80211 phy0: Atheros AR9300 Rev:4 mem=0xbc000000, irq=144
[  137.736191] PTM 1.0.27    PTM (E1) firmware version 0.30
[  137.740151] ifxmips_ptm: PTM init succeed
[  140.815288] device eth0.1 entered promiscuous mode
[  140.818833] device eth0 entered promiscuous mode
[  140.831969] br-lan: port 1(eth0.1) entered forwarding state
[  140.836268] br-lan: port 1(eth0.1) entered forwarding state
[  140.904486] IPv6: ADDRCONF(NETDEV_UP): ptm0: link is not ready
[  142.839616] br-lan: port 1(eth0.1) entered forwarding state



BusyBox v1.24.2 () built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 DESIGNATED DRIVER (Bleeding Edge, 50048)
 -----------------------------------------------------
  * 2 oz. Orange Juice         Combine all juices in a
  * 2 oz. Pineapple Juice      tall glass filled with
  * 2 oz. Grapefruit Juice     ice, stir well.
  * 2 oz. Cranberry Juice
 -----------------------------------------------------
root@OpenWrt:/# 
```

I have to modify the kernel module but I can't because these modifications are specific to my hardware. Can I add a directive to my dts file to deactivate the lantiq endianness workaround in drivers/mtd/maps/lantiq-flash.c ?